pyccl/pyutils.py

```diff
@@ -463,3 +465,26 @@ def _spline_integrate(x, ys, a, b):
         result = result[0]
 
     return result
+
+
+def _check_array_params(f_arg, arr3=False):
+    """Check whether an argument `f_arg` passed into the constructor of
+    Tracer() is valid.
+
+    If the argument is set to `None`, it will be replaced with a special array
+    that signals to the CCL wrapper that this argument is NULL.
+    """
+    if f_arg is None:
+        # Return empty array if argument is None
+        f1 = NoneArr
+        f2 = NoneArr
+        f3 = NoneArr
+    else:
+        f1 = np.atleast_1d(np.array(f_arg[0], dtype=float))
+        f2 = np.atleast_1d(np.array(f_arg[1], dtype=float))
+        if arr3:
+            f3 = np.atleast_1d(np.array(f_arg[2], dtype=float))
+    if arr3:
+        return f1, f2, f3
+    else:
+        return f1, f2
```

Comment:
How is this change related to fastpt and the current PR?

Reply:
This function has become more generally useful, so I moved it here so it can be imported from `pyutils.py`, rather than `tracers.py`